nixos/modules/virtualisation/containers.nix

```diff
@@ -123,7 +123,7 @@ let
       EXIT_ON_REBOOT=1 \
       exec ${config.systemd.package}/bin/systemd-nspawn \
         --keep-unit \
-        -M "$INSTANCE" -D "$root" $extraFlags \
+        -M "$INSTANCE" -D "$root" -U $extraFlags \
```

Comment:
I also attempted to do this, but I have at the moment several problems with systemd running in nspawn in unprivileged moment. A different problem is, that you cannot setup setuid wrappers in a user namespaces.

Reply:
User namespaces are unlikely to work for all applications. For instance, bind-mounting a host data directory into a user-namespaced container may lead to ownership and permission issues. I prefer to use user namespaces on a case-by-case basis. #35541 proposes an extraFlags option to pass additional flags like `-U` to systemd-nspawn.

Reply:
systemd-nspawn does not even start for me at the moment. An upstream patch is required.

Reply:
https://github.com/systemd/systemd/issues/6244

Reply:
Bind mouting would require shiftfs to be upstreamed in linux.

Reply:
Guess we'll have to wait then...